src/NCalc.Core/Parser/LogicalExpressionParser.cs

```diff
@@ -12,6 +12,10 @@ namespace NCalc.Parser;
 /// </summary>
 public static class LogicalExpressionParser
 {
+    // Cache for different parser configurations
+    private static readonly ConcurrentDictionary<LogicalExpressionParserOptions, Parser<LogicalExpression>> ParserCache = new();
+
+    // Legacy cache for backward compatibility
     private static readonly ConcurrentDictionary<CultureInfo, Parser<LogicalExpression>> Parsers = new();
```

Comment:
To prevent confusion with this legacy cache and avoid this additional field, what if we can convert a CultureInfo implicitly to LogicalExpressionParserOptions?